### PR TITLE
fix(angular1-router): add support for using the component helper

### LINKED ISSUE #13860: ngComponentRouter not calling $routerOnActivate in angular 1.5 when using a template URL
I know it is not released yet, but I am testing out the new router  with Angular 1.5 RC1. The router file I am using is pulled from Here: https://github.com/brandonroberts/angularjs-component-router/blob/master/lib/angular_1_router.js

The $routerONActivate hook is never called if my router directive uses a templateUrl. When I switch it to just be a template (using an ng-include pointing to the url) it gets called as expected.

### PR DESCRIPTION
**Pull request by petebacondarwin**
In Angular 1.5 there is a new helper method for creating component directives.
See https://docs.angularjs.org/guide/component for more information about components.

These kind of directives only match the `E` element form and the previously component
router only created HTML that matched directives that matched the `A` attribute form.

This commit changes the `<ng-outlet>` directive so that it generates custom HTML
elements rather divs with custom attributes to trigger the relevant component to
appear in the DOM.

Going forward, Angular 1.5 users are encouraged to create their router components
using the following style:

```
myModule.componnet('component-name', {
  // component definition object
});
```

Closes angular/angular.js#13860
Closes #5278

BREAKING CHANGE:

The component router now creates custom element HTML rather than custom attribute
HTML, in order to create a new component. So rather than

``` html
<div custom-component></div>
```

it now creates

``` html
<custom-component></custom-component>
```

If you defined you router components using the `directive()` helper and
specified the `restrict` properties such that element matching was not allowed,
e.g. `restrict: 'A'` then these components will no longer be instantiated
by the component router and the outlet will be empty.

The fix is to include `E` in the `restrict` property.

`restrict: 'EA'`

Note that this does not affect directives that did not specify the `restrict`
property as the default for this property is already `EA`.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -39,9 +39,9 @@
     "zone.js": "0.5.10"
   },
   "devDependencies": {
-    "angular": "^1.4.7",
-    "angular-animate": "^1.4.7",
-    "angular-mocks": "^1.4.7",
+    "angular": "^1.5.0-rc.2",
```
Comment (btford) [inline package.json]: we should double check that this version bump doesn't affect benchmarks, etc.

**Comment by btford** [comment]
LGTM, not sure what the circle CI failure is about.

File: package.json (hunk 1 above)
Comment (petebacondarwin) [inline package.json]: How do we do that?
Are the benchmarks run as part of the CI builds?

**Comment by alxhub** [comment]
Hi Pete, go ahead and merge this when you're ready.

**Comment by veenayp** [comment]
Could you provided latest compiled version of router i.e. .js file

**Comment by petebacondarwin** [comment]
Closed via d86be245b8c1fdbbad603d8ddeb3b5bc6278ee74

**Comment by ericj79** [comment]
Is there a way to get this updated component router through bower, for using with Angular 1.5.x?

**Comment by petebacondarwin** [comment]
The router is published via npm: `npm install @angular/router`
The version published there is not yet correct.
Look for the new version later this week.

**Comment by y4nnL** [comment]
As of today, it seems the version published with npm does not contain the fix
What is the best way to patch the directive?

**Comment by petebacondarwin** [comment]
@y4nnL : Today the npm package does not contain the fixed files. The most reliable way to get it is to build it from the Angular 2 code. We are working to get a new release published very soon.

But @andreiho has kindly built it and put it in a gist for you.

See  https://github.com/angular/angular.js/issues/12926#issuecomment-189170152

**Comment by gkalpak** [comment]
BTW, it's now available on npm (version 0.2.0).

**Pull request closed without merging**